src/Pelagos/Bundle/AppBundle/Command/ColdStorageFlagCommand.php

```diff
@@ -0,0 +1,181 @@
+     * @throws \ReflectionException If reflection class fails.
+     * @throws \Exception If stubfile not readable/accessible.
+     * @throws \Exception If infofile not readable/accessible.
+     *
+     * @return integer
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $entityManager = $this->getContainer()->get('doctrine.orm.entity_manager');
+        $entityEventDispatcher = $this->getContainer()->get('pelagos.event.entity_event_dispatcher');
+        $systemPerson = $entityManager->find(Person::class, 0);
+
+        $infoFileName = $input->getArgument('infofile');
+        if (file_exists($infoFileName)) {
+            $infoFileContents = file($infoFileName);
+            $udi = trim($infoFileContents[0]);
+
+            $nudi = preg_replace('/:/', '.', $udi);
+            $infoPath = pathinfo($infoFileName)['dirname'];
+            $size = preg_replace('/size \(bytes\): /', '', trim($infoFileContents[1]));
+            $hash = preg_replace('/orig sha256: /', '', trim($infoFileContents[2]));
+            $stubFileName = "$infoPath/$nudi-manifest.zip";
+
+            if (file_exists($stubFileName)) {
+                $output->writeln("UDI: ($udi)");
+                $output->writeln("Size: ($size)");
+                $output->writeln("Hash: ($hash)");
+                $output->writeln("stubFileName: ($stubFileName)");
+
+                $output->writeln("Attempting to flag $udi as Cold Stored.");
+
+                $datasets = $entityManager->getRepository('Pelagos\Entity\Dataset')->findBy(array('udi' => $udi));
+
+                if (count($datasets) == 0) {
+                    throw new \Exception('Could not find a dataset with the udi provided.');
+                } else {
+                    $output->writeln('Dataset Found.');
+                }
+
+                $dataset = $datasets[0];
+
+                $datasetSubmission = $dataset->getDatasetSubmission();
+                if (!($datasetSubmission instanceof DatasetSubmission)) {
+                    throw new \Exception('Could not find Dataset Submission.');
+                } else {
+                    $output->writeln('Submission Found.');
+                    // Create a new submission using latest via constructor.
+                    $newDatasetSubmission = new DatasetSubmission($datasetSubmission);
+                    if ($entityManager->contains($newDatasetSubmission)) {
+                        throw new \Exception('Attempted to create a dataset submission that is already tracked');
+                    }
+                    $output->writeln('original submission ID is: ' . $dataset->getDatasetSubmission()->getId());
+                    $output->writeln('new submission ID is initially: ' . $newDatasetSubmission->getId());
+
+                    // This method sets submission status to: DatasetSubmission::STATUS_IN_REVIEW (3).
+                    $newDatasetSubmission->setDatasetSubmissionReviewStatus();
```

Comment:
I don't think this should be done as part of this command, if you are setting this, then we should create a DatasetSubmissionReview object as well.

Reply:
This is a mistake I meant to remove earlier...good catch.